addons/ui/org.openhab.ui.cometvisu/src/main/java/org/openhab/ui/cometvisu/internal/config/ConfigHelper.java

```diff
@@ -0,0 +1,1077 @@
+                        found = true;
+                        break;
+                    }
+                }
+            }
+            if (found)
+                break;
+
+        }
+        if (!found) {
+            if (plugins == null) {
+                plugins = new Plugins();
+                pages.getMeta().getPluginsOrIconsOrMappings().add(plugins);
+            }
+            plugins.getPlugin().add(plugin);
+        }
+    }
+
+    public void addToRoot(Object root, JAXBElement<?> child) {
+        if (root instanceof Page) {
+            ((Page) root).getPageOrGroupOrNavbar().add(child);
+        } else if (root instanceof Group) {
+            ((Group) root).getPageOrGroupOrLine().add(child);
+        }
+    }
+
+    public Layout createLayout(int colspan) {
+        return createLayout(colspan, 1);
+    }
+
+    public Layout createLayout(int colspan, int rowspan) {
+        Layout layout = new Layout();
+        layout.setColspan(new BigDecimal(colspan));
+        if (rowspan != 1) {
+            layout.setRowspan(new BigDecimal(rowspan));
+        }
+        return layout;
+    }
+
+    public String getLabel(Widget widget) {
+        String label = app.getItemUIRegistry().getLabel(widget);
+        // remove format
+        label = label.replaceAll("\\[.*\\]$", "");
+        return StringEscapeUtils.escapeXml(label);
+    }
+
+    // public JAXBElement<?> convertToJAXBElement(Object bean) {
+    // return new JAXBElement(new QName(bean.getClass().getSimpleName()
+    // .toLowerCase()), bean.getClass(), bean);
+    // }
+
+    /**
+     * add the separating line in the navbar
+     *
+     * @param page
+     *            - the page the navbar should be searched in
+     * @param position
+     *            - the position of the navbar
+     * @param ifNotEmpty
+     *            - true: only add the separator ot the navbar has already
+     *            content (e.g. prevent a seperator at the beginning of the
+     *            navbar)
+     */
+    public void addSeparatorToNavbar(Page page, NavbarPositionType position, boolean ifNotEmpty) {
+        Navbar navbar = getNavbar(page, position);
+        if (navbar != null) {
+            if (!ifNotEmpty || navbar.getPageOrGroupOrLine().size() > 0) {
+                Line line = new Line();
+                line.setLayout(createLayout(0));
+                navbar.getPageOrGroupOrLine().add(factory.createNavbarLine(line));
+            }
+        }
+    }
+
+    public String getExistingIconName(String ohIcon) {
+        if (ohIcon == null || ohIcon.isEmpty())
+            return null;
+        if (hasIconMapping(ohIcon)) {
+            return getIconMapping(ohIcon);
+        } else {
+            // add the oh-icon to the icon definitions
+            if (app.getIconProvider().hasIcon(ohIcon, "classic", Format.PNG) != null) {
+                IconDefinition iconDef = factory.createIconDefinition();
+                iconDef.setUri("/images/" + ohIcon + ".png");
```

Comment:
Careful: The openHAB items are not available at '/images/x.png' anymore (well they are, but this is a temporary workaround), but at '/icons/x?format=png' now - see https://www.eclipse.org/smarthome/documentation/features/ui/icons.html#icon-servlet for the exact url.